src/verify.c

```diff
@@ -781,9 +826,18 @@ enum swupd_code verify_main(void)
 {
 	struct manifest *official_manifest = NULL;
 	int ret;
-	struct list *subs = NULL;
+	struct list *all_subs = NULL;
```

Comment:
Do we really need that many pointers?
Instead of saving everything can't we free data when we don't need anymore? And group tasks into functions so we can keep some variables local

Reply:
It is hard to reduce the number of pointers in this case, we could do it if we try hard but it would make the code even harder to read.

This verify_main function has many possible paths, so it is already a confusing function, and the main reason for this is that it includes the paths from 4 different commands, repair, diagnose and os-install, and the deprecated verify.

Maybe we should really consider splitting this function into many smaller functions so each of those commands can have their own <command>_main function instead and consume all the pieces performed in this function.

I made changes to the code in the last update to free the data as soon as we are done with it though, so at least that was improved.

Reply:
> It is hard to reduce the number of pointers in this case, we could do it if we try hard but it would make the code even harder to read.
> 
> This verify_main function has many possible paths, so it is already a confusing function, and the main reason for this is that it includes the paths from 4 different commands, repair, diagnose and os-install, and the deprecated verify.
> 
> Maybe we should really consider splitting this function into many smaller functions so each of those commands can have their own _main function instead and consume all the pieces performed in this function.
> 
> I made changes to the code in the last update to free the data as soon as we are done with it though, so at least that was improved.

@otaviobp I think that except from this ^ I finished addressing all your comments in this last update. Let me know what you think.